scripts/release/steps/update-changelog.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+const chalk = require("chalk");
+const dedent = require("dedent");
+const fs = require("fs");
+const prsMergedSince = require("prs-merged-since");
+const semver = require("semver");
+const { logPromise, keypress } = require("../utils");
+
+async function prepareReleaseNotes({ version, previousVersion }) {
+  const versionDiff = semver.diff(version, previousVersion);
+
+  let changes;
+  let warning;
+  let releaseNotes;
+
+  if (versionDiff !== "patch") {
+    // Changelog is the blog post, so we just add a link to it
+    changes = "blog post link";
+
+    const date = new Date();
+    const year = date.getFullYear();
+    const month = new String(date.getMonth() + 1).padStart(2, "0");
+    const day = new String(date.getDate()).padStart(2, "0");
+    const blogPostFile = `website/blog/${year}-${month}-${day}-${version}.md`;
+    const blogPostPath = `${year}/${month}/${day}/${version}.html`;
+
+    releaseNotes = `- [Release Notes](https://prettier.io/blog/${blogPostPath})`;
+
+    if (!fs.existsSync(blogPostFile)) {
+      warning = chalk`Make sure the blog post file {yellow ${blogPostFile}} will be created.`;
+    }
+  } else {
+    // Changelog should be the most important PRs, let's add everything
+    // and ask for the releaser to edit the file later.
+    changes = "merged PRs since last release";
+    const prs = await prsMergedSince({
```

Comment:
prs-merged-since makes a ton of GitHub API requests (when running against large repos like prettier), and GitHub limits the number of requests you can make per hour if you don't use an API Token. I think we can get through one or two runs of prs-merged-since against prettier without a token, but if someone had to run this script a couple times in a row because they were debugging stuff, they might hit the issue. prs-merged-since will read the token from the environment variable `GITHUB_API_TOKEN`; we could print a warning here if it's not set, maybe? Or maybe prs-merged-since should be the one to print the warning instead.

Reply:
I thought about that, but patch releases aren't likely to have too many PRs so I *think* we're safe.

And yes, we can add a warning, let me know what you think.

Reply:
I remembered that when using the JS API, you pass the token as `githubApiToken`, not using env.

If I remember correctly, we still have to request every PR in the repo, even for patch releases, due to a limitation of the GitHub API.

I think we should pass `process.env.GITHUB_API_TOKEN` into `prsMergedSince` here if it's present, and if it isn't, print a warning.

Reply:
Oh I'm already passing the `process.env.GITHUB_API_TOKEN`, just didn't document it anywhere.

I included the warning